wasm/wat/parser_test.go

```diff
@@ -348,7 +355,7 @@ func TestParseModule_Errors(t *testing.T) {
 		{
 			name:        "import func double result",
 			input:       "(module (import \"\" \"\" (func (param i32) (result i32) (result i32))))",
-			expectedErr: "1:55: redundant result field in module.import[0].func",
```

Comment:
saying redundant result field would be too much logic vs below. one important thing this PR does is prepares us to re-use the logic implementing the type use in imported functions for both module type and function parsing. This will be next.